recipes/TIMIT/ASR/seq2seq/hparams/train_webdataset.yaml

```diff
@@ -0,0 +1,206 @@
+# ################################
+# Model: seq2seq ASR on TIMIT with CRDNN (with LiGRU)
+# Authors:
+# * Mirco Ravanelli 2020
+# * Ju-Chieh Chou 2020
+# * Abdel Heba 2020
+# * Aku Rouhe 2021
+# ################################
+
+# Seed needs to be set at top of yaml, before objects with parameters are made
+seed: 1234
+__set_seed: !apply:torch.manual_seed [!ref <seed>]
+output_folder: !ref results/augment_noise_CRDNN_webdataset/<seed>-2
+wer_file: !ref <output_folder>/wer.txt
+save_folder: !ref <output_folder>/save
+train_log: !ref <output_folder>/train_log.txt
+
+# Data files
+data_folder: !PLACEHOLDER  # e.g. /path/to/TIMIT
+manifests:
+    train: !ref <output_folder>/train.json
+    valid: !ref <output_folder>/dev.json
+    test: !ref <output_folder>/test.json
+shards_root: !PLACEHOLDER  # Where to put the shards (top-dir), e.g. /tmp/timit-shards
+examples_per_shard: 1024
+skip_prep: False # Skip data preparation including shards
+
+
+# Training parameters
```

Comment:
More comments can be added for sure. But here I'd also like to hear your opinion, so I'll explain briefly. So the dynamic batching adds new utterances to the batch until the batch total number of elements (like `.numel()`) exceeds `target_batch_numel`, expect `max_batch_numel` should not be exceeded. Since the elements are in samples here, the numbers are large and unintuitive. However, they can be converted to seconds (like 23.75s here). 

Personally find it kind a nice to think of this dynamic batchsize as target number of seconds in a batch, do you feel the same? That's why I think this could be decent way to parametrize a dynamic batch size.

Reply:
I think it totally makes sense. This would ease the relation to VRAM. I think ESPnet does the "same" even though they use seq_len which is quite disconnected from time if you don't know the parameterization.